connectors/sources/sharepoint_online.py

```diff
@@ -879,258 +1003,240 @@ async def validate_config(self):
 
-    async def _with_list_item_access_control(
-        self, site_web_url, site_list_name, list_item
-    ):
-        if not self._dls_enabled():
-            return list_item
+        Returns:
+            dict: dictionary representing an user access control document
+            {
+                "_id": "some.user@spo.com",
+                "identity": {
+                    "email": "email:some.user@spo.com",
+                    "username": "user:some.user"
+                },
+                "_timestamp": "2023-06-30 12:00:00",
+                "query": {
+                    "template": {
+                        "params": {
+                            "access_control": [
+                                "email:some.user@spo.com",
+                                "user:some.user",
+                                "group:1234-abcd-id"
+                            ]
+                        }
+                    }
+                }
+            }
+        """
 
-        list_item_role_assignments = await self.client.site_list_item_role_assignments(
-            site_web_url, site_list_name, list_item["id"]
-        )
+        if "UserName" in user:
+            username_field = "UserName"
+        elif "userPrincipalName" in user:
+            username_field = "userPrincipalName"
+        else:
+            return
 
-        access_control = await self._access_control_for_role_assignments(
-            site_web_url, list_item_role_assignments
-        )
+        prefixed_groups = set()
 
-        return self._decorate_with_access_control(list_item, access_control)
+        async for group in self.client.groups_user_transitive_member_of(
+            user[username_field]
+        ):
+            group_id = group["id"]
+            if group_id:
+                prefixed_groups.add(_prefix_group(group_id))
 
-    def _dls_enabled(self):
-        if self._features is None:
-            return False
+        email = user.get("EMail", user.get("mail", None))
+        username = user[username_field]
 
-        if not self._features.document_level_security_enabled():
-            return False
+        prefixed_mail = _prefix_email(email)
+        prefixed_username = _prefix_user(username)
+        id_ = email if email else username
 
-        return self.configuration["use_document_level_security"]
+        access_control = list({prefixed_mail, prefixed_username}.union(prefixed_groups))
 
-    def access_control_query(self, access_control):
-        return {"query": {"template": {"params": {"access_control": access_control}}}}
+        if "Modified" in user:
+            timestamp = datetime.strptime(user["Modified"], TIMESTAMP_FORMAT)
+        else:
+            timestamp = iso_utc()
 
-    def _user_access_control_doc(self, user, access_control):
         return {
-            "_id": user.get("Id"),
+            # Here we're intentionally using the email/username without the prefix
```

Comment:
why intentionally though? What's the use-case?

Reply:
We want to query by emails/usernames of AppSearch users, which are not prefixed.

Reply:
But I think you still use `prefixed_mail` and `prefixed_username`?